File: Ex. 27.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char const *argv[])
{
    float r, n, a, b;

        printf ("Digite um valor positivo para ter sua tabuada: ");
        scanf ("%f", & n);

        while (n<0)
        {
            printf ("Valor invalido, digite um valor positivo: ");
            scanf ("%f", & n);
        }

        printf ("Digite o limite inferior da tabuada: ");
        scanf ("%f", & a);
        
        printf ("Digite o limite superior da tabuada: ");
        scanf ("%f", & b);

        while (a>=b)
        {
            printf ("Valor invalido, digite um valor maior que o limite inferior: ");
            scanf ("%f", & b);
        }

        for (size_t; b >= a; b--)
        {  
        r=b*n;
        printf ("\n %.0f x %.0f = %.0f", n, b, r);
        }
}

File: Python/Ex 27.py
n = float(input("Digite o valor para fazer a tabuada: "))

while(n <= 0):
    print("Não pode numero negativo!")
    n = int(input("Digite um outro numero para obter a tabuada: "))

b = float(input("Digite o valor inferior da tabuada: "))
a = float(input("Digite o valor superior da tabuada: "))

while(a < b):
    print("valor invalido")
    a = float(input("Digite um valor maior que o limite inferior: "))

while(b <= a):
    r = n * a
    print(f'{n} X {a} = {r}')
    a = a - 1